ruff_cgx/linter.py

```diff
@@ -5,12 +5,49 @@
 
 from .utils import (
     create_virtual_render_content,
-    get_script_range,
+    extract_script_content,
     parse_cgx_file,
     run_ruff_check,
 )
 
 
+def _prepare_content_for_linting(content: str) -> str | None:
+    """
+    Prepare CGX content for linting by extracting Python and creating virtual content.
+
+    Args:
+        content: The CGX file content as a string
+
+    Returns:
+        The prepared Python code ready for ruff, or None if no script section
+    """
+    # Parse the CGX file
+    parsed = parse_cgx_file(content)
+    if not parsed.script_node:
+        return None
+
+    # Extract pure Python content
+    script_content = extract_script_content(parsed.script_node)
+    if not script_content:
+        return None
+
+    # Prepend with comment lines to preserve line numbers for diagnostics
+    prefix_lines = ["#\n"] * script_content.start_line
+    python_lines = script_content.python_code.splitlines(keepends=True)
+    modified_lines = prefix_lines + python_lines
+
+    # Add newline to lines that don't have it
+    modified_lines = [
+        line if line.endswith("\n") else f"{line}\n" for line in modified_lines
+    ]
```

Comment:
Isn't it enough to check just the last line? What else is it ever going to do?

I would recommend to avoid all loops if you can :)